src/main/java/com/hms_networks/americas/sc/extensions/historicaldata/RapidCatchUp.java

```diff
@@ -0,0 +1,242 @@
+package com.hms_networks.americas.sc.extensions.historicaldata;
+
+import com.ewon.ewonitf.Exporter;
+import com.hms_networks.americas.sc.extensions.datapoint.DataPoint;
+import com.hms_networks.americas.sc.extensions.json.JSONException;
+import com.hms_networks.americas.sc.extensions.logging.Logger;
+import com.hms_networks.americas.sc.extensions.string.StringUtils;
+import com.hms_networks.americas.sc.extensions.system.time.SCTimeUnit;
+import com.hms_networks.americas.sc.extensions.system.time.SCTimeUtils;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.util.List;
+
+/**
+ * This class addresses the need for historical data tracking to "catch up" quickly when there are
+ * periods where the tracker is not running. For example, the Flexy is powered off for several
+ * weeks. For correct behavior, it is required that injectJvmLocalTime from {@link SCTimeUtils} is
+ * called at startup.
+ *
+ * @author HMS Networks, MU Americas Solution Center
+ * @since 1.14.2
+ */
+public class RapidCatchUp {
+
+  /**
+   * Threshold for enabling rapid catch up mode in milliseconds. If current time - start time is
+   * less than this amount, catchup mode will not be enabled.
+   */
+  public static final long RAPID_CATCH_UP_TIME_DELTA_MILLISECONDS = SCTimeUnit.HOURS.toMillis(1);
+
+  /** Starting point for request period, which changes based on previous result. */
+  // 10 minutes for the starting point
+  public static final long RAPID_CATCH_UP_TIME_BASE = SCTimeUnit.MINUTES.toMillis(10);
+
+  /**
+   * Maximum amount of time for the request period. 48 hours was used for testing. This was with a
+   * Flexy that had 100 tags at 10 seconds intervals.
+   */
+  // 24 hours is below the testing 48, but sill will provide aggressive catch up
```

Comment:
```suggestion
  /**
   * Maximum amount of time for the request period. 48 hours was used for testing. This was with a
   * Flexy that had 100 tags at 10 seconds intervals.
   * <p>
   * 24 hours is below the testing 48, but sill will provide aggressive catch up.
   * </p>
   */
```

Reply:
I dont see documenting the set value as necessary since the code self documents that. I like that the testing value is documented and it could be mentioned that the value should be less than or equal to the tested value of 48.